realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -270,4 +303,30 @@ public int hashCode() {
         return 37;
     }
 
+
+    // Helper class for keeping track of strong references to objects.
+    private static class StrongReferenceCounter<K> {
+
+        private final Map<K, Integer> references = new IdentityHashMap<K, Integer>();
```

Comment:
I am a bit uncertain about using `IdentityHashMap`. 
```
RealmXXObject obj1 = realm.where(RealmXXObject.class).findFirst();
RealmXXObject obj2 = realm.where(RealmXXObject.class).findFirst();

Observable<RealmXXObject> observable1 = obj1.asObservable();
Observable<RealmXXObject> observable2 = obj2.asObservable();
```

Since `observable2.equals(observable1) == true`, the `observable2` won't be added to the map which means it will still be GCed.